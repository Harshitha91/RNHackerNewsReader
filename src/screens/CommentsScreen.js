import React from 'react';
import { Comment } from '../components';
import { View, StyleSheet, Text, FlatList } from 'react-native';

export default function CommentsScreen({ route }) {
    const { commentsList, title, url } = route?.params ?? {};

    return (
        <View style={styles.container}>
            <View style={styles.subContainer}>
                <View style={styles.title}>
                    <Text style={styles.text}>{title ?? ''}</Text>
                </View>
                <View style={styles.itemUrl}>
                    <Text style={styles.urlText}>{url ?? ''}</Text>
                </View>
            </View>
            <FlatList
                testID={'flatlist'}
                style={styles.list}
                data={commentsList ?? []}
                keyExtractor={item => item.toString()}
                renderItem={({ item }) => {
                    return <Comment testID={'comment'} id={item} />;
                }}
            />
        </View>
    );
}
const styles = StyleSheet.create({
    container: {
        backgroundColor: 'white',
        flex: 1,
    },
    subContainer: {
        height: 120,
        backgroundColor: 'rgba(255,110,13,1)',
        opacity: 1,
    },
    itemUrl: {
        height: 20,
        left: 0,
        position: 'absolute',
        right: 0,
        top: '45%',
    },
    urlText: {
        top: '5%',
        left: 20,
        position: 'absolute',
        backgroundColor: 'transparent',
        color: 'rgba(236,236,236,1)',
        opacity: 0.8,
    },
    list: {
        height: '100%',
    },
    title: {
        height: 35,
        top: 15,
        left: 20,
        position: 'absolute',
        right: 15,
    },
    text: {
        color: 'rgba(236,236,236,1)',
    },
});
