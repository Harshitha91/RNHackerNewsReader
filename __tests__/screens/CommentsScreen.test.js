import * as React from 'react';
import CommentsScreen from '../../src/screens/CommentsScreen';
import { render } from '@testing-library/react-native';
import renderer from 'react-test-renderer';
it(`renders correctly`, () => {
    const tree = renderer.create(<CommentsScreen />);
    expect(tree).toMatchSnapshot();
});

describe('Test Comment Screen', () => {
    it('Should contain Flatlist.', async () => {
        const { getByTestId } = render(<CommentsScreen />);
        const flatlist = getByTestId('flatlist');

        expect(flatlist).toBeTruthy();
    });
});
