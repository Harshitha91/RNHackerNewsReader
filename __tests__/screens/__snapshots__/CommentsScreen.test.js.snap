// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "rgba(255,110,13,1)",
        "height": 120,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "height": 35,
          "left": 20,
          "position": "absolute",
          "right": 15,
          "top": 15,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(236,236,236,1)",
          }
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "height": 20,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": "45%",
        }
      }
    >
      <Text
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "rgba(236,236,236,1)",
            "left": 20,
            "opacity": 0.8,
            "position": "absolute",
            "top": "5%",
          }
        }
      >
        
      </Text>
    </View>
  </View>
  <RCTScrollView
    data={Array []}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "height": "100%",
      }
    }
    testID="flatlist"
    viewabilityConfigCallbackPairs={Array []}
  >
    <View />
  </RCTScrollView>
</View>
`;
